frontend/packages/console-shared/src/components/spotlight/Spotlight.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import cx from 'classnames';
+import Portal from '@console/shared/src/components/popper/Portal';
+import './Spotlight.scss';
+
+type SpotlightProps = {
+  selector: string;
+  animate?: boolean;
+  hasBackdrop?: boolean;
+};
+
+const isInViewport = (element) => {
+  const rect = element.getBoundingClientRect();
+  return (
+    rect.top >= 0 &&
+    rect.left >= 0 &&
+    rect.bottom <= (window.innerHeight || document.documentElement.clientHeight) &&
+    rect.right <= (window.innerWidth || document.documentElement.clientWidth)
+  );
+};
+
+const Spotlight: React.FC<SpotlightProps> = ({ selector, animate, hasBackdrop }) => {
+  const element = document.querySelector(selector);
+  const { height, width, top, left } = element.getBoundingClientRect();
+  const style: React.CSSProperties = {
+    top,
+    left,
+    height,
+    width,
+    ['--ocs-spotlight-top' as React.ReactText]: `${top}px`,
+    ['--ocs-spotlight-left' as React.ReactText]: `${left}px`,
+    ['--ocs-spotlight-height' as React.ReactText]: `${height}px`,
+    ['--ocs-spotlight-width' as React.ReactText]: `${width}px`,
+  };
+
+  const [clicked, setClicked] = React.useState(false);
+  React.useEffect(() => {
+    if (!isInViewport(element)) {
+      element.scrollIntoView();
+    }
+    const handleClick = () => setClicked(true);
+    document.addEventListener('mousedown', handleClick);
+    return () => {
+      document.removeEventListener('mousedown', handleClick);
+    };
+  }, [clicked, element]);
+
+  const highlight = (
+    <div
+      className={cx('ocs-spotlight', {
+        'ocs-spotlight__element-highlight-animate': animate && !clicked,
+        'ocs-spotlight__element-highlight-noanimate': !animate,
+      })}
+      style={style}
+    />
+  );
+  return (
+    <Portal>
```

Comment:
We will need to use popperjs instead so that the spotlight moves as the node underneath it moves.

Reply:
using popperjs instead of portal